packages/patternfly-4/react-charts/src/components/ChartDonutUtilization/ChartDonutUtilization.js

```diff
@@ -0,0 +1,347 @@
+   *
+   * Note: Not compatible with theme prop
+   *
+   * @example themeColor={ChartThemeColor.blue}
+   */
+  themeColor: PropTypes.string,
+  /**
+   * Specifies the theme variant. Valid values are 'dark' or 'light'
+   *
+   * Note: Not compatible with theme prop
+   *
+   * @example themeVariant={ChartThemeVariant.light}
+   */
+  themeVariant: PropTypes.string,
+  /**
+   * The dynamic portion of the chart will change colors when data reaches the given threshold. Colors may be
+   * overridden, but defaults shall be provided.
+   *
+   * @example thresholds={[{ value: 60, color: '#F0AB00' }, { value: 90, color: 'C9190B' }]}
+   */
+  thresholds: PropTypes.any,
+  /**
+   * The title for the donut chart
+   */
+  title: PropTypes.string,
+  /**
+   * Defines a horizontal shift from the x coordinate. This should not be set manually.
+   */
+  tooltipDx: PropTypes.number,
+  /**
+   * Specifies the width of the svg viewBox of the chart container. This value should be given as a
+   * number of pixels.
+   *
+   * Because Victory renders responsive containers, the width and height props do not determine the width and
+   * height of the chart in number of pixels, but instead define an aspect ratio for the chart. The exact number of
+   * pixels will depend on the size of the container the chart is rendered into.
+   *
+   * Note: The parent container must be set to the same width in order to maintain the aspect ratio. Otherwise, the
+   * innerRadius may need to be set when using this property.
+   */
+  width: PropTypes.number
+};
+
+const ChartDonutUtilization = ({
+  data = [],
+  donutDx,
+  donutOrientation = 'left',
+  legendComponent,
+  legendData,
+  legendOrientation = styles.legend.legendOrientation,
+  showStatic = true,
+  standalone = true,
+  subTitle,
+  themeColor,
+  themeVariant,
+  thresholds,
+  title,
+  tooltipDx,
+  transform,
+
+  // destructure last
+  theme = getDonutUtilizationTheme(themeColor, themeVariant),
+  donutHeight = theme.pie.height,
+  donutWidth = theme.pie.width,
+  height = theme.pie.height,
+  innerRadius = ((donutHeight || donutWidth) - 34) / 2,
+  width = theme.pie.width,
+  ...rest
+}) => {
+  // Returns computed data representing pie chart percentages
+  const getComputedData = () => {
+    const computedData = [];
+    data.forEach(datum => {
+      const i = computedData.length;
+      computedData.push({
+        x: datum.x,
+        y: i > 0 ? datum.y - data[i - 1].y : datum.y
+      });
+    });
+    if (showStatic) {
+      const yVal = data[computedData.length - 1] ? data[computedData.length - 1].y : 0;
+      computedData.push({ y: 100 - yVal });
+    }
+    return computedData;
+  };
+
+  // Returns thresholds with default color scale
+  const getDonutThresholds = () => {
+    const result = [];
+    if (thresholds) {
+      for (let i = 0; i < thresholds.length; i++) {
+        result.push({
+          color: thresholds[i].color ? thresholds[i].color : styles.thresholds.colorScale[i],
```

Comment:
If `thresholds.length` > `styles.thresholds.colorScale.length` and no color prop is provided in thresholds beyond the length of the colorScale array, the resulting threshold color will be undefined.

Reply:
Right. I meant to mod that, so the colors scales are repeated.